File: Tyche.Android/app/src/main/java/com/pipel/tyche/domain/PoolLayout.kt
package com.pipel.tyche.domain

import com.pipel.core.type.DateTime
import com.pipel.core.type.NonEmptyString
import com.pipel.core.type.Uuid

data class PoolLayout(
    val poolLayoutId: Uuid,
    val name: NonEmptyString,
    val openingStartDateTime: DateTime,
    val openingEndDateTime: DateTime
)

File: Tyche.Backend/Pipel.Tyche.PoolLayout/Pipel.Tyche.PoolLayout/Data/PoolLayout.fs
namespace Pipel.Tyche.PoolLayout.Data

open System
open MongoDB.Bson.Serialization.Attributes

type PoolLayoutEntityPK =
    { PoolLayoutId: Guid }

[<CLIMutable>]
type PoolLayoutEntity =
    { [<BsonId>]
      PoolLayoutId: Guid
      Name: string
      OpeningStartDateTime: DateTime
      OpeningEndDateTime: DateTime }
